db/db_impl/db_impl.cc

```diff
@@ -147,20 +149,21 @@ int64_t kDefaultLowPriThrottledRate = 2 * 1024 * 1024;
 
 DBImpl::DBImpl(const DBOptions& options, const std::string& dbname,
                const bool seq_per_batch, const bool batch_per_txn)
-    : env_(options.env),
-      dbname_(dbname),
+    : dbname_(dbname),
       own_info_log_(options.info_log == nullptr),
       initial_db_options_(SanitizeOptions(dbname, options)),
+      env_(initial_db_options_.env),
+      fs_(initial_db_options_.file_system),
```

Comment:
Are you guaranteed that file_system is not NULL at this point?

Reply:
Yes, SanitizeOptions() would have allocated one if the user passed in NULL.